ledger/cryptarchia-ledger/src/lib.rs

```diff
@@ -741,4 +742,82 @@ pub mod tests {
             Err(LedgerError::NullifierExists)
         ));
     }
+
+    #[test]
+    fn test_conversions_for_leader_proof() {
+        let commitment = Commitment::from([0u8; 32]);
+        let commitment_bytes: [u8; 32] = commitment.into();
+
+        let _zero_bytes = [0u8; 32];
+        assert!(matches!(commitment_bytes, _zero_bytes));
+
+        let commitment_ref = commitment.as_ref();
+        assert_eq!(commitment_ref, &_zero_bytes);
+
+        let nullifier = Nullifier::from([0u8; 32]);
+        let _nullifier_bytes: [u8; 32] = nullifier.into();
+        assert!(matches!(_nullifier_bytes, _zero_bytes));
+
+        let slot = Slot::genesis();
+        let leader_proof = LeaderProof::dummy(slot, commitment, Commitment::from([0; 32]));
+
+        assert_eq!(leader_proof.commitment(), &commitment);
+        assert_eq!(leader_proof.evolved_commitment(), &commitment);
+        assert_eq!(leader_proof.nullifier(), &nullifier);
+
+        // Test ser/de of compact representation for Nullifier
+        assert_tokens(&nullifier.compact(), &[Token::BorrowedBytes(&[0; 32])]);
+
+        // Test ser/de of compact representation for Commitment
+        assert_tokens(&commitment.compact(), &[Token::BorrowedBytes(&[0; 32])]);
+    }
+
+    #[test]
+    fn test_cryptarchia_ledger_error_cases() {
```

Comment:
can we split this in different tests cases with a more descriptive name?

Reply:
Changed since https://github.com/logos-co/nomos-node/pull/660/commits/c1ec5e54df920a1e1ec47f6eaaac51d6c44bc030 

Please LMKWYT.